### PR TITLE
Fix issue DPTP-3633 Move HyperShift control plane pods to a dedicated machine set

### PR DESCRIPTION
**Pull request by bear-redhat**
We create a dedicated machine set, and make it to be used by HyperShift control plane pods.

### CODE REVIEW

**Comment by bear-redhat** [comment]
/pjrehearse ci/rehearse/openshift-pipelines/operator/main/e2e-hypershift

**Comment by bear-redhat** [comment]
/pj-rehearse ci/rehearse/openshift-pipelines/operator/main/e2e-hypershift

**Comment by bear-redhat** [comment]
/pj-rehearse openshift-pipelines/operator/main/e2e-hypershift

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-hac-e2e-tests

**Comment by bear-redhat** [comment]
/pj-rehearse pull-ci-redhat-appstudio-infra-deployments-main-appstudio-hac-e2e-tests

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:123]: This should not be set if we have a `MachineAutoscaler` targeting this machine set.

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:136]: Does `Hypershift` set up node selector on the pods which we expect to run on the nodes from the machine set?

I guess that is what https://github.com/openshift/release/pull/43262/files#diff-6f6f9668e008a46382d5523177e1aeeb3f3d09f9c92baee608a2256b713552f6R63 is for.

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:188]:
```suggestion
```

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:190]: Do we need this?

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:160]: Do we really need a powerful type like this?

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:120]:
```suggestion
  name: hypershift-rj7rg-worker-us-east-1a
```

File: clusters/hive/hypershift.yaml
Comment (bear-redhat) [reply clusters/hive/hypershift.yaml:136]: Correct, I changed the parameter of the workflow to apply the node selector

File: clusters/hive/hypershift.yaml
Comment (bear-redhat) [reply clusters/hive/hypershift.yaml:160]: The control plane pods are CPU intensive, so I use the same machine type as we run our e2e tests.

Comment (hongkailiu) [reply clusters/hive/hypershift.yaml:159]: Where do they run before this PR? Infra nodes?

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:192]: This needs to be changed as well?
Could you please scale down and then delete `ms/hypershift-worker` after merging.

File: clusters/hive/hypershift.yaml
Comment (hongkailiu) [inline clusters/hive/hypershift.yaml:184]:
```suggestion
  name: hypershift-worker
```
```suggestion
  name: ci-dv2np-hypershift-worker-us-east-1a
```

File: clusters/hive/hypershift.yaml
Comment (bear-redhat) [reply clusters/hive/hypershift.yaml:159]: regular worker

Comment (hongkailiu) [reply clusters/hive/hypershift.yaml:159]: We had a discussion on the machineset over slack.
My concern is the cost on the more expensive machine types but you expect to reduce the number of of timeout which we do not have an effective way to monitor.
We could end up paying more for nothing.
I guess I should not block you further.

**Review by hongkailiu — APPROVED** [review]

**Comment by hongkailiu** [comment]
/hold

Free to cancel whenever you are ready to watch the rolling.

**Pull request merged**
